exporter/exporter_test.go

```diff
@@ -2244,3 +2244,112 @@ func TestProcessSentinelSlaves(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPScrapeWithPasswordFile(t *testing.T) {
```

Comment:
This is a great test, thanks for adding that.

Two comments:
1) you can shorten the list of expected values for success to maybe just `test_up 1` and `uptime_in_seconds`, that should be enough. You can also remove all the csk and cntk stuff, 

2) can you add a test that missing the password from the password file fails? 
There's a second password protected Redis instance in the test setup, you can try to connect to that and expect the exporter to return `test_up 0`.